cinema/views.py

```diff
@@ -118,12 +142,21 @@
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(viewsets.GenericViewSet,
+                   mixins.ListModelMixin,
+                   mixins.CreateModelMixin):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
+
+    def get_permissions(self):
+        if self.action == "create":
+            return (IsAuthenticated(),)
+        return super().get_permissions()
```

Comment:
In the `get_permissions` method, you are overriding the default permission classes for the 'create' action. Ensure that this behavior is intended, as it changes the permissions dynamically based on the action.